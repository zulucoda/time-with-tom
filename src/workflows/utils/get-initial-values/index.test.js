import getInitialValues from '.';

describe('Workflow - Utils - Get Initial Values - Unit Test', function () {
  test('should return merged object literal', () => {
    const workflowA = () => ({
      component: () => 'some react component A',
      payload: (values) => ({ ...values }),
      validate: (values) => ({ ...values }),
      initialValues: {
        valA1: 'valA1',
        valA2: 'valA2',
      },
    });
    const workflowB = () => ({
      component: () => 'some react component B',
      payload: (values) => ({ ...values }),
      validate: (values) => ({ ...values }),
      initialValues: {
        valB1: 'valB1',
        valB2: 'valB2',
      },
    });
    const workflows = [workflowA(), workflowB()];
    const act = getInitialValues(workflows);
    expect(act).toMatchSnapshot({
      valA1: 'valA1',
      valA2: 'valA2',
      valB1: 'valB1',
      valB2: 'valB2',
    });
  });
});
