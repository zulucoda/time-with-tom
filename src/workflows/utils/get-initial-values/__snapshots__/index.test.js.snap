// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflow - Utils - Get Initial Values - Unit Test should return merged object literal 1`] = `
Object {
  "valA1": "valA1",
  "valA2": "valA2",
  "valB1": "valB1",
  "valB2": "valB2",
}
`;
